tcks/microprofile-context-propagation/pom.xml

```diff
@@ -13,6 +13,10 @@
     <artifactId>quarkus-tck-microprofile-context-propagation</artifactId>
     <name>Quarkus - TCK - MicroProfile Context Propagation</name>
 
+    <properties>
+        <microprofile-context-propagation-tck.version>1.0.2</microprofile-context-propagation-tck.version>
```

Comment:
This one was on 1.0.1 before, but in BOM we have 1.2. But if I update the tck to 1.2, I am seeing the following failure:
```
[ERROR] Tests run: 81, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 12.23 s <<< FAILURE! - in TestSuite
[ERROR] org.eclipse.microprofile.context.tck.ManagedExecutorTest.propagateApplicationContext  Time elapsed: 0.005 s  <<< FAILURE!
java.lang.IllegalStateException: Missing propagated provider type: Application
        at io.smallrye.context.SmallRyeContextManager.getProviderPlan(SmallRyeContextManager.java:159)
```
Known problem or oversight?

Reply:
/cc @FroMage @manovotn

Reply:
Could be related to changes done in https://github.com/eclipse/microprofile-context-propagation/issues/192?
TBF I don't even know if we do any such thing as `Application` context propagation.

Reply:
> TBF I don't even know if we do any such thing as `Application` context propagation.

Ok, who can answer that question? If we don't, we should probably bump to 1.2 and exclude the test?

Reply:
Some `grep`ing and test searching shows me that Quarkus has following context providers:

* `io.quarkus.arc.runtime.context.ArcContextProvider`
* `io.quarkus.resteasy.common.runtime.ResteasyContextProvider`
* `io.quarkus.undertow.runtime.ServletThreadContextProvider` 
* `io.smallrye.context.jta.context.propagation.JtaContextProvider`

And there are no additional ones under [smallrye project](https://github.com/smallrye/smallrye-context-propagation) itself so I'd say we just don't have it.

I also found this comment from @FroMage which says the same (but is year old!) - https://github.com/eclipse/microprofile-context-propagation/pull/194#issuecomment-656002514

Looks like @Ladicek has been part of that discussion as well so he might know something I don't.

But I guess it boils down to whether we want application context propagation for Quarkus? In the meantime, I think we could update the dependency and exclude the test @famod.

Reply:
I think MP ConProp got somewhat clarified and even has a TCK test that if application context is propagated, then TCCL is propagated. But it's been a while.

I still believe Quarkus should propagate TCCL as part of the application context propagation.

Reply:
Can you file an issue @Ladicek for that please? If I do it I might write something stupid because CP is not really my field of expertise.
I would then bump the TCK and exclude that one test with a reference to that new ticket.

Reply:
If it is generic enough, we could theoretically implement this directly in Smallrye CP and just reuse it here - which is what happens with `JtaContextProvider`.

In fact, I am not sure SR CP now passes the TCKs, need to try that.

Reply:
Hmm, SR project is running 1.2 TCK version as it should and it passes the test without having application context provider. I am puzzled :-D

Reply:
I was being blind, some debugging showed me that SR impl actually has this class - [`io.smallrye.context.application.context.propagation.ApplicationContextProvider`](https://github.com/smallrye/smallrye-context-propagation/blob/main/application/src/main/java/io/smallrye/context/application/context/propagation/ApplicationContextProvider.java).

So including that in Quarkus can be as easy as this - https://github.com/quarkusio/quarkus/compare/main...manovotn:srCpAppContextProvider?expand=1
@famod feel free to add that commit^^ to your PR. I haven't tested if anything breaks by adding that, just that it fixes the test in question. I guess CI can tell us more :-)

Reply:
@manovotn there is a different failure with your commit and the tck updated to 1.2:
```
[ERROR] Tests run: 81, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 12.197 s <<< FAILURE! - in TestSuite
[ERROR] org.eclipse.microprofile.context.tck.cdi.JTACDITest.testTransactionPropagation  Time elapsed: 0.013 s  <<< FAILURE!
java.util.concurrent.CompletionException: java.util.ServiceConfigurationError: org.eclipse.microprofile.context.spi.ThreadContextProvider: io.smallrye.context.application.context.propagation.ApplicationContextProvider not a subtype
```

Reply:
Good catch, I can see that locally. No idea why yet; that context provider is a subtype of `ThreadContextProvider` from what I can tell.

Reply:
That test triggers a discovery of thread context providers and it seems like it is a `forEach` invoked on `ServiceLoader<ThreadContextProvider>` that throws the exception.

I can see the discovery being triggered for at least two different tests and each time a different CL is used to perform service loading. That could be the issue but I have zero idea why this happens or what triggers different CL usage.
`SmallRyeContextPropagationProcessor` seems to recognize all provides just right...

Reply:
Looks like a classloader issue. We've never needed to propagate the TCCL manually in Quarkus, so I would not include the `ApplicationContextProvider` by default. Not sure why the TCK would force us to do that. I'd just make that TCK faliure ignored.

Reply:
We have actually stumbled upon app context propagation issues when we discussed using `ForkJoinPool` (and particularly the default one). Since FJP messes with classloaders, my opinion is that the correct way of using FJP in Quarkus is using context propagation (which requires propagating TCCL).

Reply:
Yea but this brings the usual CP downside - since Quarkus propagated all contexts by default, we would add this overhead to all scenarios which is IMO not worth it.

Reply:
Propagating TCCL adds virtually zero overhead to what's already there, so I'm not sure I buy that argument :-)

Reply:
If you have the use case then by all means create an issue, link these two together and let's solve it there :-)
I just know we are not exactly in favor of MP CP as such, so I am not sure we want to add to it. That's all I am saying...

Reply:
So I have updated the tck to 1.2, excluded `propagateApplicationContext`, was able to remove two older exclusions and made the suite more robust to test additions or removals.